core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Player.php

```diff
@@ -2081,9 +2081,16 @@ public function giveItem(SR_Item $item)
 		return $this->sr4_inventory->addItem($item);
 	}
 
-	public function itemAmountChanged(SR_Item $item, $amount_change, $modify=true)
-	{
-		$this->sr4_inventory->itemAmountChanged($item, $amount_change);
+	public function itemAmountChanged(SR_Item $item, $amount_change, $modify=true, $use_mount=false)
```

Comment:
If `modify()` is indeed not needed for changes to mount, then it is probably nicer to have a separate `mountItemAmountChanged()` or something.

Reply:
A separate method is imho always preferable to a boolean argument switching the implementation.
Especially if the method only switches on that boolean argument.

But then again, this code is a mess already :)

Reply:
"a mess", that's why I didn't bother to nicen things up for a fix.We all know the code is unmaintainable.
And we all should know default parameters are bad and bite you sooner or later, but it's already in use, so I added another default parameter.

Reply:
resolved with new commit